src/Cache/SimpleCacheClient.php

```diff
@@ -2,43 +2,30 @@
 
 namespace Momento\Cache;
 
-use Momento\Cache\Errors\InvalidArgumentError;
+use Momento\Auth\ICredentialProvider;
 
 class SimpleCacheClient
 {
 
     private _ScsControlClient $controlClient;
     private _ScsDataClient $dataClient;
     /**
-     * @param string $authToken: momento JWT
+     * @param ICredentialProvider $authProvider: Momento credential provider
      * @param int $defaultTtlSeconds: Default Time to Live for the item in Cache
+     * @param ?int $dataClientOperationTimeoutMs: msecs after which requests should be cancelled due to timeout
      */
-    function __construct(string $authToken, int $defaultTtlSeconds, ?int $dataClientOperationTimeoutMs=null)
-    {
-        $payload = $this->parseAuthToken($authToken);
-        $this->controlClient = new _ScsControlClient($authToken, $payload["cp"]);
+    function __construct(
+        ICredentialProvider $authProvider, int $defaultTtlSeconds, ?int $dataClientOperationTimeoutMs=null
+    ) {
+        $this->controlClient = new _ScsControlClient($authProvider->getAuthToken(), $authProvider->getControlEndpoint());
         $this->dataClient = new _ScsDataClient(
-            $authToken, $payload["c"], $defaultTtlSeconds, $dataClientOperationTimeoutMs
+            $authProvider->getAuthToken(),
+            $authProvider->getCacheEndpoint(),
+            $defaultTtlSeconds,
+            $dataClientOperationTimeoutMs
```

Comment:
Is there a standard `TimeSpan`-like construct in PHP? Would be strong to avoid hoisting number conversions on the users

Reply:
I'm +1 to this idea, but if such a thing exists in PHP, we can handle it as a separate ticket/PR.